openshift_performance/ose3_perf/scripts/build_test.py

```diff
@@ -215,6 +215,11 @@ def run_builds(executor, all_builds):
 
         logger.info("All threads started, starting builds")
         wait(futures)
+
+        with ThreadPoolExecutor(max_workers=global_config["worker"]) \
+                as executor1:
+            check_build_status(executor1)
```

Comment:
Two risks that I can think of for now:
1. the threads used by this script could be out of control again
2. could have multiple threads checking on the status of the same build (no lock there) and our data-structure of storing status is not thread-safe

Reply:
Not sure I understand the concerns - will contact you offline.   check_build_status is still running in the main thread, just a bit earlier.  

Previously it was the checkpoint in the build iteration loop - each iteration waited for the previous iteration's builds to complete.   

With this change each batch waits for the previous batch to complete - if batching (-b) is not used then it becomes the checkpoint for each iteration.